modules/visual_script/editor/visual_script_property_selector.cpp

```diff
@@ -55,679 +70,1200 @@ void VisualScriptPropertySelector::_sbox_input(const Ref<InputEvent> &p_ie) {
+
+	if (clear_text) {
+		search_box->set_text("");
 	}
+	search_box->select_all();
+
+	search_visual_script_nodes->set_pressed(true);
+	search_classes->set_pressed(false);
+	search_methods->set_pressed(false);
+	search_operators->set_pressed(false);
+	search_signals->set_pressed(false);
+	search_constants->set_pressed(false);
+	search_properties->set_pressed(false);
+	search_theme_items->set_pressed(false);
+
+	scope_combo->select(0); //id0 = "Search Related" //id2 = "Search Base" //id3 = "Search Inheriters" //id4 = "Search Unrelated" //id5 "Search All"
+
+	results_tree->clear();
+	show_window(.5f);
+	search_box->grab_focus();
+	_update_results();
+}
+
+void VisualScriptPropertySelector::select_from_instance(Object *p_instance, const bool p_connecting, bool clear_text) {
+	set_title(TTR("Select from instance"));
+	base_type = p_instance->get_class();
 
-	if ((properties || seq_connect) && visual_script_generic) {
-		get_visual_node_names("", Set<String>(), found, root, search_box);
+	const Ref<Script> &p_script = p_instance->get_script();
+	if (p_script == nullptr) {
+		base_script = "";
+	} else {
+		base_script = p_script->get_path().lstrip("res://").quote(); // filepath to EditorHelp::get_doc_data().name
 	}
 
-	TreeItem *selected_item = search_options->search_item_text(search_box->get_text());
-	if (!found && selected_item != nullptr) {
-		selected_item->select(0);
-		found = true;
+	type = Variant::NIL;
+	connecting = p_connecting;
+
+	if (clear_text) {
+		search_box->set_text(" ");
 	}
+	search_box->select_all();
+
+	search_visual_script_nodes->set_pressed(false);
+	search_classes->set_pressed(false);
+	search_methods->set_pressed(true);
+	search_operators->set_pressed(false);
+	search_signals->set_pressed(true);
+	search_constants->set_pressed(true);
+	search_properties->set_pressed(true);
+	search_theme_items->set_pressed(false);
 
-	get_ok_button()->set_disabled(root->get_first_child() == nullptr);
+	scope_combo->select(2); //id0 = "Search Related" //id2 = "Search Base" //id3 = "Search Inheriters" //id4 = "Search Unrelated" //id5 "Search All"
+
+	results_tree->clear();
+	show_window(.5f);
+	search_box->grab_focus();
+	_update_results();
 }
 
-void VisualScriptPropertySelector::create_visualscript_item(const String &name, TreeItem *const root, const String &search_input, const String &text) {
-	if (search_input.is_empty() || text.findn(search_input) != -1) {
-		TreeItem *item = search_options->create_item(root);
-		item->set_text(0, text);
-		item->set_icon(0, vbc->get_theme_icon(SNAME("VisualScript"), SNAME("EditorIcons")));
-		item->set_metadata(0, name);
-		item->set_metadata(1, "action");
-		item->set_selectable(0, true);
-		item->set_collapsed(true);
-		item->set_selectable(1, false);
-		item->set_selectable(2, false);
-		item->set_metadata(2, connecting);
+void VisualScriptPropertySelector::select_from_visual_script(const Ref<Script> &p_script, bool clear_text) {
+	set_title(TTR("Select from visual script"));
```

Comment:
Maybe the "Create Visual Script Node" title sounds better (just to be consistent with "Create Shader Node" in visual shaders).

Reply:
"Create Visual Script Node" is not accurate, it allows you to select a node compatible with the visual script you are editing.
In my mind, Visual script nodes are things like sequences or conditions similar to keywords in GDScript.